### PR TITLE
NEVISACCESSAPP-4833: Check user enrolment during registration

### PR DESCRIPTION
**Pull request by laszlo-domonkos**

### CODE REVIEW

File: app/src/main/java/ch/nevis/exampleapp/common/settings/Settings.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package ch.nevis.exampleapp.common.settings
+
+/**
+ * Interface declaration of application settings. It provides application level settings properties.
+ */
+interface Settings {
```
Comment (laszlo-domonkos) [inline app/src/main/java/ch/nevis/exampleapp/common/settings/Settings.kt:6]: AllowClass2Sensors flag is needed for proper biometric authenticator enrollment checking, so I introduced this interface and its implementation that provides only this flag at this moment.

File: app/src/main/java/ch/nevis/exampleapp/domain/util/Authenticator.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    return when (val userEnrollment = userEnrollment()) {
+        is UserEnrollment.OsUserEnrollment -> {
+            if (allowClass2Sensors) {
+                userEnrollment.isEnrolledWithClass2OrClass3Sensor
+            } else {
+                userEnrollment.isEnrolled
+            }
+        }
+        is UserEnrollment.SdkUserEnrollment -> userEnrollment.isEnrolled(username)
+        else -> throw IllegalStateException("Unknown UserEnrollment object.")
```
Comment (laszlo-domonkos) [inline app/src/main/java/ch/nevis/exampleapp/domain/util/Authenticator.kt:25]: I'm not sure it is the best solution to throw an `IllegalStateException` in this case, but It should never happen.

File: app/src/main/java/ch/nevis/exampleapp/ui/selectAuthenticator/model/AuthenticatorItem.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    /**
+     * The flag that tells whether the user already enrolled the authenticator or not.
+     */
+    val isUserEnrolled: Boolean
+) {
+    /**
+     * String resource identifier of the title of the authenticator.
+     */
+    @StringRes
+    val titleResId = authenticatorTitles[aaid] ?: throw IllegalArgumentException("Unknown AAID.")
```
Comment (laszlo-domonkos) [inline app/src/main/java/ch/nevis/exampleapp/ui/selectAuthenticator/model/AuthenticatorItem.kt:32]: I'm not sure it is the best solution to throw an `IllegalArgumentException` in this case, but once again It should never happen.

**Review by tamas-toth — APPROVED** [review]

**Review by josu-vergara-lecue — APPROVED** [review]

File: app/src/main/java/ch/nevis/exampleapp/domain/util/Authenticator.kt [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package ch.nevis.exampleapp.domain.util
```
Comment (tamas-toth) [inline app/src/main/java/ch/nevis/exampleapp/domain/util/Authenticator.kt:1]: Missing header.

File: app/src/main/java/ch/nevis/exampleapp/common/settings/SettingsImpl.kt [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package ch.nevis.exampleapp.common.settings
```
Comment (tamas-toth) [inline app/src/main/java/ch/nevis/exampleapp/common/settings/SettingsImpl.kt:1]: Missing header.

File: app/src/main/java/ch/nevis/exampleapp/common/settings/Settings.kt [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package ch.nevis.exampleapp.common.settings
```
[the remark this answers is not in the document]
Comment (laszlo-domonkos) [reply app/src/main/java/ch/nevis/exampleapp/common/settings/Settings.kt:7]: Fixed.

File: app/src/main/java/ch/nevis/exampleapp/common/settings/SettingsImpl.kt (hunk 5 above)
Comment (laszlo-domonkos) [reply app/src/main/java/ch/nevis/exampleapp/common/settings/SettingsImpl.kt:7]: Fixed.

File: app/src/main/java/ch/nevis/exampleapp/domain/util/Authenticator.kt (hunk 4 above)
Comment (laszlo-domonkos) [reply app/src/main/java/ch/nevis/exampleapp/domain/util/Authenticator.kt:7]: Fixed.

File: app/src/main/java/ch/nevis/exampleapp/ui/selectAuthenticator/model/AuthenticatorItem.kt [hunk 7]
```diff
@@ -0,0 +1,54 @@
+package ch.nevis.exampleapp.ui.selectAuthenticator.model
```
[the remark this answers is not in the document]
Comment (laszlo-domonkos) [reply app/src/main/java/ch/nevis/exampleapp/ui/selectAuthenticator/model/AuthenticatorItem.kt:7]: Fixed.

**Review by tamas-toth — APPROVED** [review]

**Pull request merged**